### DISCUSSION TITLE
Alternative to deprecated $wrapNodes ?

### DISCUSSION THREAD

**Discussion by matisszemturis** [facebook/lexical#5600] [Ideas]
I'm looking for a way to wrap nodes with a specific type of node - Selection node/Linters/Highlight etc.
I find it very odd that in Playground when creating a comment, highlight box is created outside editor and then hovered over by calculating intial position. 

Before there was a $wrapNodes function, but it got deprecated.
Regarding wrapping selection - Link and Mark nodes use completely different logics, which makes it even more confusing on how to handle a seperate use case with a custom node.

Is there any solution that works for all use cases and is it even planned?

**Comment by przemodeveloper** [comment]
@matisszemturis 
You can use $setBlocksType, like `$setBlocksType(selection, () => $createHeadingNode('h1'));` (from @lexical/selection)

**Reply by ogadra** [reply]
@przemodeveloper Thanks a lot! This comment helped me!
I think it would be even more helpful if this information were added as a comment regarding `$wrapNodes` in the `@lexical/selection/range-selection.d.ts`  file.

**Comment by jazoom** [comment]
Might be worth actually mentioning this replacement [here](https://lexical.dev/docs/api/modules/lexical_selection#wrapnodes).

**Reply by etrepum** [reply]
Would be a good thing to make a PR for
